lib/ex_aws/config/auth_cache.ex

```diff
@@ -72,14 +70,35 @@ defmodule ExAws.Config.AuthCache do
           auth
 
         adapter ->
-          adapter.adapt_auth_config(auth, profile, expiration)
+          attempt_credentials_refresh(adapter, auth, profile, expiration)
       end
 
+    Process.send_after(self(), {:refresh_awscli_config, profile, expiration}, expiration)
     :ets.insert(ets, {{:awscli, profile}, auth})
 
     auth
   end
 
+  defp attempt_credentials_refresh(adapter, auth, profile, expiration, retries \\ 6) do
+    case adapter.adapt_auth_config(auth, profile, expiration) do
+      {:error, error} ->
+        if retries == 1 do
+          Process.send_after(self(), {:refresh_awscli_config, profile, expiration}, expiration)
+
+          raise "Could't get credentials from auth adapter after 6 retries, last error was #{inspect error}"
+        else
+          Process.sleep(:rand.uniform(5_000))
+          attempt_credentials_refresh(adapter, auth, profile, expiration, retries - 1)
+        end
+
+      # Always store a map on AuthCache
+      auth when is_map(auth) ->
+        auth
+
+      _unexpected -> %{}
```

Comment:
Should we ever hit this? If so, in what case?
If it's not a thing we should ever hit, and we don't know what to do if we do, then I think it's sensible to leave out this block and let it crash, since otherwise we're leaving the system in an undefined state.